time_stamper_cv/src/detector.cpp

```diff
@@ -0,0 +1,27 @@
+#include "detector.h"
+#include "ros/ros.h"
+
+detector::detector(cv::SimpleBlobDetector::Params params) {
+  detector_ = cv::SimpleBlobDetector::create(params);
+}
+
+void detector::process(const cv::Mat &input_mat) {
+  detector_->detect(input_mat, keypoints_);
+}
+
+bool detector::isKeypointsEmpty() {
+  return keypoints_.empty();
+}
+
+void detector::pollKeyPointStatus(const std::function<void(std::string)>& function) {
```

Comment:
why not just return a string?

Reply:
Because the std::function is intended to only get called if the keypoint state change.